### PR TITLE
Add eslint and run integration tests in CI

### PR DESCRIPTION
**Pull request by ClementWalter**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build-related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe):

# What is the current behavior?

No eslint configuration is enforced in the project (for the integrations tests).
The integration are not run in the CI

# What is the new behavior?

An eslint configuration has been added, some errors fixed and eslint is enforced in the CI.
The integration tests are run in the CI.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

# Other information

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,67 +11,86 @@ on:
...
+        run: rustup show
+      - name: Set-Up
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y clang llvm libudev-dev protobuf-compiler
+      - uses: Swatinem/rust-cache@v2
+      - name: Check Build for Benchmarking
+        run: |
+          pushd crates/node
+          cargo check --features=runtime-benchmarks --release
```
Comment (LucasLvy) [inline .github/workflows/test.yml:96]: You can remove this completely

Comment (LucasLvy) [inline .github/workflows/test.yml:96]: you can remove this completely

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "eslint.workingDirectories": ["tests"]
```
Comment (EvolveArt) [inline .vscode/settings.json:2]: shouldnt be here no?

Comment (ClementWalter) [reply .vscode/settings.json:2]: what do you mean? like gitignored ? this is useful for all vscode users I guess

File: .github/workflows/test.yml (hunk 1 above)
Comment (ClementWalter) [reply .github/workflows/test.yml:96]: you mean remove the whole check job?

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**